packages/@aws-cdk/aws-codedeploy/lib/lambda/deployment-group.ts

```diff
@@ -0,0 +1,196 @@
+
+import { CfnDeploymentGroup } from '../codedeploy.generated';
+import { AutoRollbackConfig, DeploymentOption, DeploymentType } from '../config';
+import { deploymentGroupNameToArn, renderAlarmConfiguration, renderAutoRollbackConfiguration } from '../utils';
+import { ILambdaApplication, LambdaApplication } from './application';
+import { LambdaDeploymentConfig } from './deployment-config';
+
+export interface ILambdaDeploymentGroup extends cdk.IConstruct {
+  readonly application: ILambdaApplication;
+  readonly deploymentGroupName: string;
+  readonly deploymentGroupArn: string;
+
+  export(): LambdaDeploymentGroupImportProps;
+}
+
+export interface LambdaDeploymentGroupProps {
+  application: LambdaApplication;
+  deploymentGroupName?: string;
+  deploymentConfig: LambdaDeploymentConfig;
+
+  alarms?: cloudwatch.Alarm[];
+
+  serviceRole?: iam.Role;
+  alias: lambda.Alias;
+
+  preHook?: lambda.IFunction;
+  postHook?: lambda.IFunction;
+
+  /**
+   * Whether to continue a deployment even if fetching the alarm status from CloudWatch failed.
+   *
+   * @default false
+   */
+  ignorePollAlarmsFailure?: boolean;
+
+  /**
+   * The auto-rollback configuration for this Deployment Group.
+   */
+  autoRollback?: AutoRollbackConfig;
+}
+export class LambdaDeploymentGroup extends cdk.Construct implements ILambdaDeploymentGroup {
+  /**
+   * Import an Lambda Deployment Group defined either outside the CDK,
+   * or in a different CDK Stack and exported using the {@link #export} method.
+   *
+   * @param parent the parent Construct for this new Construct
+   * @param id the logical ID of this new Construct
+   * @param props the properties of the referenced Deployment Group
+   * @returns a Construct representing a reference to an existing Deployment Group
+   */
+  public static import(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupImportProps): ILambdaDeploymentGroup {
+    return new ImportedLambdaDeploymentGroup(scope, id, props);
+  }
+
+  public readonly application: ILambdaApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+
+  constructor(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupProps) {
+    super(scope, id);
+
+    let serviceRole: iam.Role | undefined = props.serviceRole;
+    if (serviceRole) {
+      if (serviceRole.assumeRolePolicy) {
+        serviceRole.assumeRolePolicy.addStatement(new iam.PolicyStatement()
+          .addAction('sts:AssumeRole')
+          .addServicePrincipal('codedeploy.amazonaws.com'));
+      }
+    } else {
+      serviceRole = new iam.Role(this, 'ServiceRole', {
+        assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com')
+      });
+    }
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('arn:aws:s3:::*/CodeDeploy/*')
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('*')
```

Comment:
Is that supposed to be a star resource?

If this is what the docs say, add a link as a reference.

Reply:
This is a narrow re-write of the recommended policy: arn:aws:iam::aws:policy/service-role/AWSCodeDeployRoleForLambda, which is as follows:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": [
                "cloudwatch:DescribeAlarms",
                "lambda:UpdateAlias",
                "lambda:GetAlias",
                "sns:Publish"
            ],
            "Resource": "*",
            "Effect": "Allow"
        },
        {
            "Action": [
                "s3:GetObject",
                "s3:GetObjectVersion"
            ],
            "Resource": "arn:aws:s3:::*/CodeDeploy/*",
            "Effect": "Allow"
        },
        {
            "Action": [
                "s3:GetObject",
                "s3:GetObjectVersion"
            ],
            "Resource": "*",
            "Condition": {
                "StringEquals": {
                    "s3:ExistingObjectTag/UseWithCodeDeploy": "true"
                }
            },
            "Effect": "Allow"
        },
        {
            "Action": [
                "lambda:InvokeFunction"
            ],
            "Resource": "arn:aws:lambda:*:*:function:CodeDeployHook_*",
            "Effect": "Allow"
        }
    ]
}
```

Reply:
This didn't work out .. after some testing i decided to use the managed policy. Server deployment does the same and despite writing the policy out exactly, it still failed to deploy. This works fine and follows the recommendations. One thing I did change was grant invoke access directly on the pre and post hooks so you don't have to prefix their nme with `CodeDeployHook_`